mlir/utils/performance/common/common_utils.h

```diff
@@ -0,0 +1,152 @@
+
+  llvm::cl::opt<uint64_t> gemmK{"k", llvm::cl::desc("K dimennsion of gemm()"),
+                                llvm::cl::value_desc("positive integer"),
+                                llvm::cl::Required};
+
+  llvm::cl::opt<uint64_t> gemmN{"n", llvm::cl::desc("N dimennsion of gemm()"),
+                                llvm::cl::value_desc("positive integer"),
+                                llvm::cl::Required};
+
+  llvm::cl::opt<bool> transposeA{
+      "transA", llvm::cl::desc("whether matrix A is GxMxK (default) or GxKxM."),
+      llvm::cl::init(false)};
+
+  llvm::cl::opt<bool> transposeB{
+      "transB", llvm::cl::desc("whether matrix B is GxKxN (default) or GxNxK."),
+      llvm::cl::init(false)};
+
+  llvm::cl::opt<int> kernelRepeats{
+      "kernel-repeats",
+      llvm::cl::desc("Number of times to run the rocblas kernel"),
+      llvm::cl::value_desc("positive integer"), llvm::cl::init(1)};
+
+  llvm::cl::opt<DataType> dataType{
+      "t", llvm::cl::desc("Data type"),
+      llvm::cl::values(clEnumValN(DataType::F32, "f32", "32-bit float"),
+                       clEnumValN(DataType::F16, "f16", "16-bit float"),
+                       clEnumValN(DataType::BF16, "bf16", "bfloat16"),
+                       clEnumValN(DataType::I8, "i8", "8-bit integer")),
+      llvm::cl::Required};
+
+  llvm::cl::opt<std::string> operation{
+      "operation", llvm::cl::desc("Operation (ignored, for MLIR compat)"),
+      llvm::cl::init("gemm")};
+
+  llvm::cl::opt<std::string> arch{"arch",
+                                  llvm::cl::desc("Arch (ignored, MLIR compat)"),
+                                  llvm::cl::init("gemm")};
+  llvm::cl::opt<std::string>
+      perfConfig("perf_config",
+                 llvm::cl::desc("Perf config (ignored, MLIR compat)"),
+                 llvm::cl::init(""));
+};
+
+#define HIP_ABORT_IF_FAIL(expr)                                                \
+  do {                                                                         \
+    hipError_t error = expr;                                                   \
+    if (error != hipSuccess) {                                                 \
+      fprintf(stderr, "HIP error %s at %s:%d in %s\n",                         \
+              hipGetErrorString(error), __FILE__, __LINE__, #expr);            \
+      exit(1);                                                                 \
+    }                                                                          \
+  } while (0)
+
+static llvm::APFloat::Semantics getLlvmFltSemantics(DataType dataType) {
+  switch (dataType) {
+  case DataType::F32:
+    return llvm::APFloat::S_IEEEsingle;
+  case DataType::F16:
+    return llvm::APFloat::S_IEEEhalf;
+  case DataType::BF16:
+    return llvm::APFloat::S_BFloat;
+  case DataType::I8:
+    assert(0 && "Can't have i8 floats");
+  }
+}
+
+static void *allocAndFill(DataType dataType, size_t byteSize, bool isOut) {
```

Comment:
I think it would be good to have docs for common utils: here and other functions.

Reply:
... also, why are we putting a bunch of code into header files? Can we make a `benchmarkUtils.cpp` that both the rocblas and ck benchmark drivers link to? (Like, obviously the `#define` needs to be in a header but still)

Or at least if you're going to do this sort of thing, `.cpp.inc` would make it clearer what the deal is.